### PR TITLE
boards: factorize some nucleo f4/f7/l4 i2c configuration

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

During I2C rework, I noticed that the I2C configurations were very similar between nucleo boards. This PR is factorizing the I2C config of 3 of them (based on F4): nucleo-f446re, nucleo-f446ze and nucleo-f429zi.
This PR is only applied to them because the I2C configuration depends on the clock configuration and the inclusion of the header file containing the common i2c config must be put after the one containing the common clock config. Otherwise, the inclusion will have to be in the middle of the `periph_conf.h` file and I don't know if reviewers will accept this.

The name of the I2C common header file is built using the following template:
`cfg_i2c_<i2c device number>_<scl pin number>_<sda pin number>`
Better suggestions are welcome.

### Issues/PRs references

None

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Tested on [nucleo-f429zi](https://gist.github.com/MrKevinWeiss/a26d2e5d4a27f8f52f9ab539479ddd1f) and it passes.  I don't have the other boards on hand.

File: boards/nucleo-f429zi/include/periph_conf.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 #include "periph_cpu.h"
 #include "f4/cfg_clock_168_8_1.h"
+#include "cfg_i2c_1_8_9.h"
```
[the remark this answers is not in the document]
Comment (aabadie) [reply boards/nucleo-f429zi/include/periph_conf.h]: @PeterKietzmann, the naming scheme applied in the current status of this PR is provided in the initial comment. @vincent-d proposed to also use the port which is a good idea.
There's no real convention, I'm just following what was already done with clock configurations and spi divtables by @haukepetersen but applied to I2C configuration.

**Comment by aabadie** [comment]
Sorry, accidentally dropped the remote branch.

**Comment by aabadie** [comment]
@vincent-d, @MrKevinWeiss, I reworked this PR to adapt more F4 based nucleo boards. I'll test on the modified boards I have now (f410, f411, f429, f412, f446) and report the results later.

**Comment by aabadie** [comment]
I could only test on nucleo-f410rb/f429zi/f412zg and all worked. For the others, I have problems flashing the boards and I don't know why this is happening...

**Comment by aabadie** [comment]
Just in case, I've adapted and tested this PR on some nucleo F7 and L4 boards.

**Comment by MrKevinWeiss** [comment]
So it looks like the F410rb is about 100 kHz and the L476rg is 40 kHz, this is probably because the CLOCK_APB1_DIV for the L476 is div 4 and others are div 2.  I also don't think everything scales depending on the core clock selected.  I believe #9804 addresses that.

**Comment by aabadie** [comment]
Thanks for reviewing and testing @MrKevinWeiss!

**Pull request merged**